fl4health/clients/fed_prox_client.py

```diff
@@ -108,14 +121,14 @@ def compute_loss(
             target: (torch.Tensor): Ground truth data to evaluate predictions against.
 
         Returns:
-            Losses: Object containing checkpoint loss, backward loss and additional losses indexed by name.
-            Additional losses includes proximal loss.
+            TrainingLosses: an instance of TrainingLosses containing checkpoint loss, backward loss and
+                additional losses indexed by name. Additional losses includes proximal loss.
         """
-        loss = self.criterion(preds["prediction"], target)
-        proximal_loss = self.get_proximal_loss()
-        total_loss = loss + proximal_loss
-        losses = Losses(checkpoint=loss, backward=total_loss, additional_losses={"proximal_loss": proximal_loss})
-        return losses
+        loss, additional_losses = self.compute_loss_and_additional_losses(preds, features, target)
+        total_loss = loss + additional_losses["proximal_loss"]
+        # adding the vanilla loss to the additional losses to be used by update_after_train
+        additional_losses["loss"] = loss
```

Comment:
Nice catch!

Reply:
In this example, I wonder if it makes more sense to just override `compute_training_loss` and `compute_evaluation_loss` themselves such that the body of 
`compute_training_loss`
```
loss = self.criterion(preds["prediction"], target)
proximal_loss = self.get_proximal_loss()
additional_losses = {"proximal_loss": proximal_loss, "loss": loss}
return TrainingLosses(backward=loss + proximal_loss, additional_losses=additional_losses)
```
and the body of 
`compute_evaluation_loss`
```
loss = self.criterion(preds["prediction"], target)
return EvaluationLosses(checkpoint=loss, additional_losses={})
```
The reason is that proximal loss doesn't really make sense for the validation set (it's always zero). Just a thought.

Reply:
That makes sense, will do.

Reply:
I did it a bit differently. please check.

Reply:
Looks good to me.